mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java

```diff
@@ -410,7 +409,7 @@ public FlagsUpdated(MailboxSession.SessionId sessionId, User user, MailboxPath p
     class Added extends MetaDataHoldingEvent {
         private final Map<MessageUid, MessageMetaData> added;
 
-        public Added(MailboxSession.SessionId sessionId, User user, MailboxPath path, MailboxId mailboxId, SortedMap<MessageUid, MessageMetaData> uids) {
+        public Added(MailboxSession.SessionId sessionId, User user, MailboxPath path, MailboxId mailboxId, Map<MessageUid, MessageMetaData> uids) {
```

Comment:
are you sure it doesn't break anything ?
It doesn't look like something related to serialization

Reply:
I was in the hard time to convert a scala map to sorted map. I will revert old code

Reply:
so hard, any one know how to convert a scala map to a java TreeMap? :(

Reply:
ugly but seem to work
```scala
new JavaTreeMap[MessageUid, JavaMessageMetaData](added.mapValues(_.toJava).asJava)
```